src/atomvm_dialyzer_provider.erl

```diff
@@ -0,0 +1,331 @@
+        end,
+        Warnings
+    ),
+    ok.
+
+%% @private
+do_dialize(Config, State) ->
+    %% check base_plt exista and is up to date
+    BasePLT = base_plt_absname(Config),
+    case dialyzer:plt_info(BasePLT) of
+        {ok, _} ->
+            check_base_plt(Config);
+        _ ->
+            do_build_base_plt(Config)
+    end,
+
+    %% dialyze sources
+    PLT = plt_absolute_name(State),
+    case dialyzer:plt_info(PLT) of
+        {ok, _} ->
+            check_app_plt(Config, State);
+        _ ->
+            do_build_plt(Config, State)
+    end,
+    AppBEAMs = get_app_beam_abspath(State),
+    try
+        dialyzer:run([
+            {analysis_type, succ_typings}, {files_rec, [AppBEAMs]}, {plts, [PLT, BasePLT]}
+        ])
+    of
+        [] ->
+            rebar_api:notice("No problems found by dialyzer.~n"),
+            ok;
+        Problems ->
+            print_warnings(Problems)
+    catch
+        throw:{dialyzer_error, Reason} ->
+            rebar_api:error("Dialyzer failed! reason: ~p.~n", [Reason])
+    end,
+    ok.
+
+% @private
+check_base_plt(Config) ->
+    PLT = base_plt_absname(Config),
+    BEAMdir = get_base_beam_path(Config),
+    try
+        dialyzer:run([
+            {analysis_type, plt_check}, {plts, [PLT]}, {output_plt, PLT}, {files_rec, BEAMdir}
+        ])
+    of
+        [] ->
+            ok;
+        _ ->
+            do_build_base_plt(Config)
+    catch
+        throw:{dialyzer_error, _} ->
+            do_build_base_plt(Config)
+    end,
+    ok.
+
+% @private
+base_plt_absname(Config) ->
+    Home =
+        case os:getenv("HOME") of
+            false ->
+                rebar_api:error("Unable to locate users home directory");
+            Path ->
+                string:trim(Path)
+        end,
+    Base = maps:get(base_plt_location, Config, filename:join(Home, ".cache/rebar3")),
+    [Version, _] = string:split(string:trim(os:cmd("atomvm -version")), "+"),
+    BasePLT = filename:absname_join(filename:absname(Base), "AtomVM-" ++ Version ++ ".plt"),
+    rebar_api:debug("Base PLT file: ~p", [BasePLT]),
+    string:trim(BasePLT).
+
+% @private
+do_build_base_plt(Config) ->
+    %% build plt
+    BEAMdir = get_base_beam_path(Config),
+    PLT = base_plt_absname(Config),
+    try
+        dialyzer:run([
+            {analysis_type, plt_build}, {init_plt, PLT}, {output_plt, PLT}, {files_rec, BEAMdir}
+        ])
+    of
+        [] ->
+            ok;
+        Failure ->
+            print_warnings(Failure),
+            rebar_api:error("Failed to create project plt!~n")
+    catch
+        throw:{dialyzer_error, Error} ->
+            rebar_api:error("Failed to crete plt, error:~p~n", [Error])
+    end,
+    ok.
+
+% @private
+check_app_plt(Config, State) ->
+    PLT = plt_absolute_name(State),
+    try dialyzer:run([{analysis_type, plt_check}, {plts, [PLT]}]) of
+        _ ->
+            ok
+    catch
+        throw:{dialyzer_error, _} ->
+            do_build_plt(Config, State)
```

Comment:
Why is this commented?

Reply:
I believe that was just leftover dead code I forgot to remove, it's removed now.